### DISCUSSION TITLE
Read a xport file in parallel in read_file_multiprocessing()

### DISCUSSION THREAD

**Discussion by guoranba** [Roche/pyreadstat#229] [Q&A]
As per the specification in pyreadstat’s documentation, Xport file is not supported as it does not have the number of rows recorded in the metadata, which is needed for function read_file_multiprocessing. I'm wondering if it works when providing an extra parameter and setting it to the number of rows which can be obtained in other ways.

**Comment by ofajardo** [comment]
Unfortunately it is not possible at the moment.

**Comment by ofajardo** [comment]
I have done a new release 1.2.2 where now you have the argument num_rows that has to be equal or larger than the number of rows in the file. You have to determine this number yourself, either reading the file wihthout multiprocessing first, reading in the original application, or for example setting a very large number, if then the number of rows is less than your number, that means all rows were read, otherwise you have to repeat the operation until the number of rows is less than your number.

**Reply by guoranba** [reply]
Really appriciate it! I will test and get back to you soon.
